rust-runtime/aws-smithy-types/src/lib.rs

```diff
@@ -314,43 +266,50 @@ impl TryFrom<Number> for f32 {
                 if v <= (1 << 24) {
                     Ok(v as Self)
                 } else {
-                    Err(Self::Error::U64ToFloatLossyConversion(v))
+                    Err(TryFromNumberErrorKind::U64ToFloatLossyConversion(v).into())
                 }
             }
             Number::NegInt(v) => {
                 if (-(1 << 24)..=(1 << 24)).contains(&v) {
                     Ok(v as Self)
                 } else {
-                    Err(Self::Error::I64ToFloatLossyConversion(v))
+                    Err(TryFromNumberErrorKind::I64ToFloatLossyConversion(v).into())
                 }
             }
-            Number::Float(v) => Err(Self::Error::F64ToF32LossyConversion(v)),
+            Number::Float(v) => Err(TryFromNumberErrorKind::F64ToF32LossyConversion(v).into()),
         }
     }
 }
 
 #[cfg(test)]
 mod number {
     use super::*;
+    use crate::error::{TryFromNumberError, TryFromNumberErrorKind};
 
     macro_rules! to_unsigned_converter_tests {
         ($typ:ident) => {
             assert_eq!($typ::try_from(Number::PosInt(69u64)).unwrap(), 69);
 
             assert!(matches!(
                 $typ::try_from(Number::PosInt(($typ::MAX as u64) + 1u64)).unwrap_err(),
-                TryFromNumberError::OutsideIntegerRange(..)
+                TryFromNumberError {
+                    kind: TryFromNumberErrorKind::OutsideIntegerRange(..)
+                }
```

Comment:
I think we should consider creating constructors for error like this:
```rust
TryFromNumberError::outside_integer_range(..)
```
I don't think it should happen in this PR.